python/cudf/cudf/core/frame.py

```diff
@@ -91,7 +91,7 @@ def _num_rows(self) -> int:
 
     @property
     def _column_names(self) -> List[Any]:  # TODO: List[str]?
-        return self._data.names
+        return list(self._data.names)
```

Comment:
We previously returned a `tuple` while hinting that we were returning a `list`. Which one  should we return?

Reply:
From the perspective of immutability, a `tuple`. However, I have come across a few developer mistakes where the result of this property was used as if it were a `list`, perhaps because of the incorrect typing. cc: @isVoid who ran into this with me in this commit: https://github.com/rapidsai/cudf/pull/10123/commits/a4d88c39959572731250cb92bca5260a37685c69